Tests/ImageFetcherTests/ImageFetcherTests.swift

```diff
@@ -39,4 +39,29 @@ final class ImageFetcherTests: XCTestCase {
         let actual = fetcher.taskCount
         XCTAssertEqual(expected, actual)
     }
+
+    func testContinuationsAreNotLeaked() async throws {
+        let session = URLSession(configuration: .mock)
+        MockURLProtocol.responseDelay = 5.0
+        MockURLProtocol.responseProvider = { url in
+            (Color.random().image(CGSize(width: 100, height: 100)).pngData()!, Mock.makeResponse(url: url))
+        }
+
+        let cache = MockCache(onData: { _ in throw MockCache.CacheError(reason: "File missing") })
+        let sut = ImageFetcher(cache, session: session)
+
+        let url = URL(string: "https://example.com")!
+        async let result = await sut.load(url)
+
+        try await Task.sleep(nanoseconds: NSEC_PER_SEC)
+        sut.cancel(url)
```

Comment:
Shorter delays resulted in `cancel(_:)` being called in tests before the `ImageFetcherTask` had been added to `tasks`